crates/interpreter/src/valid.rs

```diff
@@ -436,9 +435,8 @@ impl SideTable {
             unsupported(if_debug!(Unsupported::SideTable))
         })?;
         let pop_cnt = Self::pop_cnt(source, target)?;
-        let entry = &mut self.entries[source.side_table];
-        assert!(entry.is_none());
-        *entry = Some(SideTableEntryView { delta_ip, delta_stp, val_cnt, pop_cnt });
+        self.entries[source.side_table] =
```

Comment:
Let's keep some debug assertions.
```suggestion
        debug_assert!(self.entries[source.side_table].is_invalid());
        self.entries[source.side_table] =
```

Where `fn is_invalid(self) -> bool` checks that the `u64` is zero. Also we should probably add `fn invalid() -> Self` that returns zero (instead of relying on Default, which we should remove if possible).

Zero is by luck a good sentinel value because it represents the maximum negative IP and STP jump.